### DISCUSSION TITLE
Adding WebSocket support Montoya

### DISCUSSION THREAD

**Discussion by chopicalqui** [PortSwigger/burp-extensions-montoya-api#15] [Ideas]
Dear PortSwigger Team

I am the author of the [Turbo Data Miner](https://portswigger.net/bappstore/84350b8f090a4388a9d12cca141f201b) extension and I am super excited about the new API.

I reviewed the new API and what I am looking for is Montoya's support for WebSockets. I would love having the following WebSocket capabilities:

- Access to the already collected WebSocket data (similar to interface method `proxy.Proxy.history()`)
- Applying custom scan checks on WebSockets (similar to interface `scanner.ScanCheck`)
- Sending our own WebSockets (similar to interface method `http.Http.issueRequest()`)

Looking forward to the discussion.
Cheers!

**Comment by chopicalqui** [comment]
Hello Again.
Just updated Burp Suite Professional and realised that WebSocket support has been added.
Amazing thanks!

**Comment by SeanBurnsUK** [comment]
Thanks for the feedback. We do have more web socket work planned.

**Comment by swindlesmccoop** [comment]
I got a notification that I was mentioned in this discussion?

**Reply by SeanBurnsUK** [reply]
Sorry, Spam message which has been deleted.

**Reply by swindlesmccoop** [reply]
Thanks, I was really confused. Sorry to be a bother.

**Comment by Hannah-PortSwigger** [comment]
Further support for WebSocket functionality has been added in our latest Early Adopter release.

You can check out our release notes here: https://portswigger.net/burp/releases/professional-community-2023-2

**Comment by Abb4d0n** [comment]
I've been looking at the WebSocket functionality and it's good to see basic support.
However, I did not see a way to add custom tabs for WebSocket messages, or am I missing something?
It would be great to have this functionality to write extensions that can visualize decoded WS messages.

**Reply by Hannah-PortSwigger** [reply]
Thanks for the feedback! Unfortunately, we've not added that functionality. However, as you mention, it would be extremely useful.

We'll get a feature request for this raised.
